### PR TITLE
workflows: Fix libclang-abi test after update to use download-artifac…

### LINKED ISSUE #64852: [Infrastructure] ABI tester flaky for release builds
Almost half of the PR's filed for the release fails in the abi tester right now, it's a little scary to not have that test working when we pass 17.0.0-final. See logs here for example: https://github.com/llvm/llvm-project-release-prs/actions/runs/5879937279/job/15947702167?pr=595

It seems like @tstellar is out right now, @nikic / @kwk do you know anything about how that abi tester works or why it seems broken?

### PR DESCRIPTION
**Pull request by tstellar**
…t v3

### CODE REVIEW

**Comment by tstellar** [comment]
I'm going to create another pull request against the release/17.x branch to test this.

**Comment by tstellar** [comment]
Test run is here: https://github.com/llvm/llvm-project/actions/runs/5931195646

**Review by nikic — APPROVED** [review]
LGTM, this is in line with the recommendation at https://github.com/actions/download-artifact#compatibility-between-v1-and-v2v3.

**Review by tru — APPROVED** [review]

File: .github/workflows/libclang-abi-tests.yml [hunk 1]
```diff
@@ -146,10 +146,12 @@ jobs:
         uses: actions/download-artifact@v3
         with:
           name: build-baseline
+          path: build-baseline
       - name: Download latest
         uses: actions/download-artifact@v3
         with:
           name: build-latest
+          path: build-latest
```
Comment (tstellar) [inline .github/workflows/libclang-abi-tests.yml:154]: Test Comment.

Comment (tstellar) [reply .github/workflows/libclang-abi-tests.yml:154]: Test Comment 2.

**Comment by tstellar** [comment]
Test Comment.

**Comment by tstellar** [comment]
Test Comment.

**Pull request merged**